solr/bin/solr

```diff
@@ -138,7 +138,7 @@ if [[ $? -ne 0 ]] ; then
   echo >&2 "${PATH}"
   exit 1
 else
-  JAVA_VER_NUM=$(echo $JAVA_VER | head -1 | awk -F '"' '/version/ {print $2}' | sed -e's/^1\.//' | sed -e's/[._-].*$//')
+  JAVA_VER_NUM=$(expr match "${JAVA_VER}" '.*java version.*[0-9]\{1,3\}\.\([0-9]\{1,3\}\)\.[0-9]\{1,3\}_[0-9]\{1,3\}.*')
```

Comment:
:question: The old expression allows versions without the update number suffix (`_[0-9]{1,3}`). Should versions without this suffix be permitted or not?